sceptre_cdk_handler/cdk.py

```diff
@@ -147,58 +153,126 @@ def bootstrapless_config(self) -> dict:
         """
         return self.arguments.get('bootstrapless_config', {})
 
+    @property
+    def path_is_to_cdk_json(self) -> bool:
+        return self.cdk_template_path.name.lower() == 'cdk.json'
+
+    @property
+    def path_is_to_python_file(self) -> bool:
+        return self.cdk_template_path.suffix == '.py'
+
+    @property
+    def stack_logical_id(self) -> Optional[str]:
+        return self.arguments.get('stack_logical_id')
+
+    def validate(self):
+        self._check_prerequisites()
+        if self.path_is_to_cdk_json:
+            self._check_cdk_json()
+        elif not self.path_is_to_python_file:
+            raise TemplateHandlerArgumentsInvalidError(
+                "The path argument must either reference a Python file with a CDK Stack class or a "
+                "cdk.json file at the root of a CDK project."
+            )
+
+        if self.deployment_type == 'bootstrapped' and self.bootstrapless_config:
+            raise TemplateHandlerArgumentsInvalidError(
+                "You cannot specify a bootstrapless_config with the bootstrapped deployment_type"
+            )
+        elif self.deployment_type == "bootstrapless" and self.bootstrap_qualifier:
+            raise TemplateHandlerArgumentsInvalidError(
+                "You cannot specify a bootstrap_qualifier with the bootstrapless deployment_type"
+            )
+        super().validate()
+
+    def _check_cdk_json(self):
+        if self.cdk_context and any(isinstance(v, (list, dict)) for v in self.cdk_context.values()):
+            raise TemplateHandlerArgumentsInvalidError(
+                "You cannot use nested values within your CDK context when your path is to a cdk.json "
+                "file. If you need to specify such values, put them in the context of your cdk.json "
+                "file."
+            )
+        if self.stack_logical_id is None:
+            raise TemplateHandlerArgumentsInvalidError(
+                "You must specify the stack_logical_id of the stack in your app in order to use the "
+                "use a cdk.json file with the CDK handler."
```

Comment:
Actually I'm going to keep this one the way it is. The reason why is because with this Template handler, you could actually use Sceptre to deploy a _number_ of CDK projects. So there might not be just _one_ cdk.json.

Reply:
I see what you're saying. I'll rephrase this.

Reply:
I've rephrased this as "No stack_logical_id has been specified. In order to use a cdk.json as you template, "you must indicate the logical id of the stack you want to deploy from your CDK app."